.travis.yml

```diff
@@ -1,28 +1,68 @@
+matrix:
+  include:
+    - os: linux
+      compiler: gcc
+    - os: osx
+      osx_image: xcode7.3
+      compiler: clang
+
 language: cpp
 
-compiler:
-  - gcc
+cache:
+  directories:
+    - ${HOME}/local
+    - ${HOME}/cpanm
 
 addons:
   hosts:
     - 127.0.0.1.xip.io
     - alternate.127.0.0.1.xip.io
 
 before_install:
-  # upgrade g++ and libstdc++ to build nghttp2
-  - sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
-  - sudo apt-add-repository --yes ppa:smspillaz/cmake-2.8.12
-  - sudo apt-get --yes update
-  - sudo apt-get install --yes cmake cmake-data g++-4.8 libstdc++-4.8-dev php5-cgi wget
-  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
-  - $CXX --version
-  # for speed, pre-install deps installed in `before_script` section as ubuntu packages
-  - sudo apt-get install -qq cpanminus libipc-signal-perl liblist-moreutils-perl libwww-perl libio-socket-ssl-perl zlib1g-dev
+  - source ./misc/travis/export_env
+
+  # get latest version of cpanminus
+  - |
+    if [[ ! -e "${INSTALL_DIR}/bin/cpanm" ]]; then
+      curl -L https://cpanmin.us/ --create-dirs -o ${INSTALL_DIR}/bin/cpanm
+      chmod +x ${INSTALL_DIR}/bin/cpanm
+    fi
+  - |
+    case "${TRAVIS_OS_NAME}" in
+      "linux" )
+        # upgrade g++ and libstdc++ to build nghttp2
+        sudo apt-add-repository --yes ppa:ubuntu-toolchain-r/test
+        sudo apt-add-repository --yes ppa:smspillaz/cmake-2.8.12
+        sudo apt-get -qq update
+        sudo apt-get install -qq cmake cmake-data g++-4.8 libstdc++-4.8-dev php5-cgi wget
+        if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
+        $CXX --version
+        # for speed, pre-install deps installed in `before_script` section as ubuntu packages
+        sudo apt-get install -qq libipc-signal-perl liblist-moreutils-perl libwww-perl libio-socket-ssl-perl zlib1g-dev
+        ;;
+      "osx"   )
+        brew update
+        ls -al ${INSTALL_DIR}/lib/
+        # install openssl-1.0.2
+        if [[ ! -e "${INSTALL_DIR}/lib/libssl.a" ]]; then
+          curl -L https://www.openssl.org/source/openssl-1.0.2h.tar.gz | tar xzf -
+          (cd openssl-1.0.2h && ./Configure darwin64-x86_64-cc --prefix=${INSTALL_DIR} no-shared && make && make install)
+        fi
+        # install zlib-1.2.8
+        if [[ ! -e "${INSTALL_DIR}/lib/libz.a" ]]; then
+          curl -L http://zlib.net/zlib-1.2.8.tar.gz | tar xzf -
+          (cd zlib-1.2.8 && ./configure --prefix=${INSTALL_DIR} --static && make && make install)
+        fi
```

Comment:
I think it's better not to use homebrew where we can build from source, because
* as a premise, caching homebrew libraries seems to be heavy job (re-installing homebrew is required I believe)
* so, building from source once and reusing it is faster than installing via homebrew every time
* as additional bonus, by doing this, we can use the same versions in all environments, and don't have to care anymore how to install specific version which doesn't exist in homebrew repository